include/openthread/message.h

```diff
@@ -69,6 +69,19 @@ typedef enum otMessagePriority
     OT_MESSAGE_PRIORITY_HIGH   = 2, ///< High priority level.
 } otMessagePriority;
 
+/**
+ * Defines the OpenThread message origins.
+ *
+ */
+typedef enum otMessageOrigin
+{
+    OT_MESSAGE_ORIGIN_THREAD_NETIF = 0, ///< Message originates from Thread Netif.
+    OT_MESSAGE_ORIGIN_HOST_DISALLOW_LOOPBACK =
+        1,                                     ///< Message originates from host and should not be passed back to host.
+    OT_MESSAGE_ORIGIN_HOST_ALLOW_LOOPBACK = 2, ///< Message originates from host and can be passed back to host.
+    OT_MESSAGE_ORIGIN_HOST_UNTRUSTED      = 3, ///< Message originates from an untrusted source on host.
+} otMessageOrigin;
```

Comment:
Just a suggestion, using shorter comments so it all fit in single line (looks nicer)
```suggestion
typedef enum otMessageOrigin
{
    OT_MESSAGE_ORIGIN_THREAD_NETIF = 0,           ///< Originates from Thread Netif.
    OT_MESSAGE_ORIGIN_HOST_DISALLOW_LOOPBACK = 1, ///< Originates from host but should not be passed back to host.
    OT_MESSAGE_ORIGIN_HOST_ALLOW_LOOPBACK = 2,    ///< Originates from host and can be passed back to host.
    OT_MESSAGE_ORIGIN_HOST_UNTRUSTED      = 3,    ///< Originates from an untrusted source on host.
} otMessageOrigin;
```